src/pages/index.js

```diff
@@ -12,54 +11,140 @@ import {
   profileEditForm,
   addCardButton,
   addCardForm,
+  changeAvatarButton,
+  changeAvatarForm,
 } from "../utils/constants.js";
 import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import ModalWithForm from "../components/ModalWithForm.js";
 import ModalWithImage from "../components/ModalWithImage.js";
 import UserInfo from "../components/UserInfo.js";
 import Section from "../components/Section.js";
+import Api from "../components/API.js";
+
+//API
+const api = new Api({
+  baseURL: "https://around-api.en.tripleten-services.com/v1",
+  authToken: "ea161ff4-cda4-4a20-a8db-64fec38336d8",
+});
+
+api
+  .getAppInfo()
+  .then(([userData, cards]) => {
+    user.setUserInfo({
+      name: userData.name,
+      bio: userData.about,
+    });
+    user.changeAvatarImage(userData.avatar);
+    section.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error(`Failed to load app info: ${err}`);
+  });
 
 //instantiate
 const section = new Section(
   {
-    items: initialCards,
     renderer: (data) => {
       section.addItem(createCard(data));
     },
   },
   cardSelectors.cardListEl,
 );
 
-const newCardModal = new ModalWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit,
-);
+const newCardModal = new ModalWithForm({
+  modalSelector: "#add-card-modal",
+  handleFormSubmit: (data) => {
+    newCardModal.renderLoading(true);
+    api
+      .addCard({
+        name: data.title,
+        link: data.link,
+      })
+      .then((data) => {
+        section.addItem(createCard(data));
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        newCardModal.renderLoading(false);
+        newCardModal.close();
```

Comment:
`   newCardModal.close();`

* You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well

**Please, correct it everywhere**